tests/autologging/test_autologging_utils.py

```diff
@@ -642,3 +645,130 @@ def test_autologging_event_logger_default_impl_warns_for_log_autolog_called_with
+        ("sklearn", "0.20.2", False),
+        ("pytorch-lightning", "1.0.5", True),
+        ("pytorch-lightning", "1.0.4", False),
+    ],
+)
+@mock.patch(
+    "mlflow.utils.autologging_utils._module_version_info_dict", _module_version_info_dict_patch
+)
+def test_is_autologging_integration_supported(flavor, module_version, expected_result):
+    module_name, _ = _cross_tested_flavor_to_module_name_and_module_key[flavor]
+    with mock.patch(module_name + ".__version__", module_version):
+        assert expected_result == _is_autologging_integration_supported(flavor)
+
+
+@mock.patch(
+    "mlflow.utils.autologging_utils._module_version_info_dict", _module_version_info_dict_patch
+)
+def test_disable_for_unsupported_versions_warning_sklearn_integration():
+    log_warn_fn_name = "mlflow.utils.autologging_utils._logger.warning"
+
+    def is_sklearn_warning_fired(log_warn_fn_args):
+        return 'You are using an unsupported version of' in log_warn_fn_args[0][0] and \
+            log_warn_fn_args[0][1] == 'sklearn'
+
+    with mock.patch("sklearn.__version__", "0.20.3"):
+        AUTOLOGGING_INTEGRATIONS.clear()
+        with mock.patch(log_warn_fn_name) as log_warn_fn:
+            mlflow.autolog(disable_for_unsupported_versions=True)
+            assert all(not is_sklearn_warning_fired(args) for args in log_warn_fn.call_args_list)
+        with mock.patch(log_warn_fn_name) as log_warn_fn:
+            mlflow.autolog(disable_for_unsupported_versions=False)
+            assert all(not is_sklearn_warning_fired(args) for args in log_warn_fn.call_args_list)
+
+        with mock.patch(log_warn_fn_name) as log_warn_fn:
+            mlflow.sklearn.autolog(disable_for_unsupported_versions=True)
+            assert len(log_warn_fn.call_args_list) == 0
+        with mock.patch(log_warn_fn_name) as log_warn_fn:
+            mlflow.sklearn.autolog(disable_for_unsupported_versions=False)
+            assert len(log_warn_fn.call_args_list) == 0
+
+    with mock.patch("sklearn.__version__", "0.20.2"):
+        AUTOLOGGING_INTEGRATIONS.clear()
+        with mock.patch(log_warn_fn_name) as log_warn_fn:
+            mlflow.autolog(disable_for_unsupported_versions=True)
+            assert all(not is_sklearn_warning_fired(args) for args in log_warn_fn.call_args_list)
+        with mock.patch(log_warn_fn_name) as log_warn_fn:
+            mlflow.autolog(disable_for_unsupported_versions=False)
+            assert any(is_sklearn_warning_fired(args) for args in log_warn_fn.call_args_list)
+
+        with mock.patch(log_warn_fn_name) as log_warn_fn:
+            mlflow.sklearn.autolog(disable_for_unsupported_versions=True)
+            assert len(log_warn_fn.call_args_list) == 0
+        with mock.patch(log_warn_fn_name) as log_warn_fn:
+            mlflow.sklearn.autolog(disable_for_unsupported_versions=False)
+            assert len(log_warn_fn.call_args_list) == 1 and \
+                is_sklearn_warning_fired(log_warn_fn.call_args_list[0])
```

Comment:
```suggestion
            assert log_warn_fn.call_count == 1 and is_sklearn_warning_fired(log_warn_fn.call_args)
```

- https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.call_args
- https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.call_count